lib/charms/tls_certificates_interface/v1/tls_certificates.py

```diff
@@ -441,60 +471,69 @@ def restore(self, snapshot: dict):
         self.expiry = snapshot["expiry"]
 
 
-class CertificateExpiredEvent(EventBase):
-    """Charm Event triggered when a TLS certificate is expired."""
-
-    def __init__(self, handle: Handle, certificate: str):
-        super().__init__(handle)
-        self.certificate = certificate
-
-    def snapshot(self) -> dict:
-        """Returns snapshot."""
-        return {"certificate": self.certificate}
-
-    def restore(self, snapshot: dict):
-        """Restores snapshot."""
-        self.certificate = snapshot["certificate"]
-
-
-class CertificateRevokedEvent(EventBase):
-    """Charm Event triggered when a TLS certificate is revoked."""
+class CertificateInvalidatedEvent(EventBase):
+    """Charm Event triggered when a TLS certificate is invalidated."""
 
     def __init__(
         self,
         handle: Handle,
+        reason: Literal["expired", "revoked"],
         certificate: str,
-        certificate_signing_request: str,
-        ca: str,
-        chain: List[str],
-        revoked: bool,
+        certificate_signing_request: Optional[str] = None,
+        ca: Optional[str] = None,
+        chain: Optional[List[str]] = None,
+        revoked: bool = False,
```

Comment:
I think this is redundant with the `reason` (and it probably was already redundant with the name of the event previously, my bad!)

Reply:
Agree, actually I wasn't event using it. I was confusing it with the `revoked` in the jsonschema. Removed.